api/apps/geoprocessing/src/migrations/geoprocessing/1655819134885-AddsPuvsprCalculationsTable.ts

```diff
@@ -0,0 +1,20 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddPuvsprCalculationsTable1655819134885
+  implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+      CREATE TABLE "puvspr_calculations"(
+        "id" uuid PRIMARY KEY DEFAULT gen_random_uuid(),
+        "project_id" uuid NOT NULL,
+        "feature_id" uuid NOT NULL,
+        "pu_id" integer NOT NULL,
```

Comment:
can we add a FK to `projects_pu.id` with cascade so that whenever the corresponding `projects_pu` row is deleted we also get the relevant `puvspr_calculations` rows deleted, without having to add another task to the GC code to clean up this table when the parent is deleted?

(if keeping as is - and in any case we should do this in a separate PR, if going with the FK - then we should also update the regular cleanups on project deletion to also GC the orphaned rows from this table - again, if at all, in a separate PR)

Reply:
So instead of using `pu_id integer`, you would prefer using the `id` of the `projects_pu` table? If it is a better solution we can do it, in case we go with this change, when generating puvspr.dat file we would need to make an extra query for getting the `integer puids` for both legacy and marxan projects, also this change will affect a little bit the export and import of this data.